tests/functional/test_fusion_hw.py

```diff
@@ -0,0 +1,115 @@
+# -*- coding: utf-8 -*-
```

Comment:
It does not make sense to add tests for a deprecated module. Deprecation is only a mechanism to not break existing functionality before users had chance to migrate off that module, the code will not be further developed and will be removed, so spending additional work on it is wasted in my eyes.

Reply:
Additionally the extra code is something you have to take in consideration in any further tasks, this thankfully does not impact Python as it is a fully dynamic and does not have compile time checks which would require fixes.

Reply:
> It does not make sense to add tests for a deprecated module. Deprecation is only a mechanism to not break existing functionality before users had chance to migrate off that module, the code will not be further developed and will be removed, so spending additional work on it is wasted in my eyes.

I think that while this module is still part of the collection it should be tested. Also, it didn't require much time from my side.

> Additionally the extra code is something you have to take in consideration in any further tasks, this thankfully does not impact Python that much as it is a fully dynamic and does not have compile time checks which would require fixes, but the tests still get run and have to pass.

If it was integration test or e2e test, yes. This runs almost instantly.

Reply:
It is not about running time but rather about maintenance burden, the effort is usually better spent elsewhere. There is always other work that needs to be done and perfectionism rarely pays off :)

Reply:
Hmm, you're right. The maintenance doesn't worth it.